### PR TITLE
Enable flake8 warnings for all currently non-violated rules

### PR DESCRIPTION
**Pull request by practicalswift**
* Enable `flake8` warnings for all currently non-violated rules
* Fix accidental redefinition via list comprehension

### CODE REVIEW

**Comment by practicalswift** [comment]
This should probably be reviewed by @jnewbery :-)

**Comment by jnewbery** [comment]
This is an improvement in style, but it doesn't change behaviour (`tx` is not used after this line)

I'm not sure if it makes sense to open PRs to fix individual flake8 violations. By my count, there are currently >5000:

```
→ flake8 test/functional/*py | wc -l
5101
```

Fixing them in PRs like this will generate a huge amount of review work.

**Comment by practicalswift** [comment]
@jnewbery Agreed – fixing all `flake8` violations would clearly be a waste of time.

And yes, we were lucky this time that `tx` was never used again after the redefinition. But why count on luck? .-)

Redefinitions via list comprehensions is a real and recurring gotcha. Enabling it in `python-lint.sh` and thus making sure we're never bitten by it again seems like an easy win to me :-)

**Comment by jnewbery** [comment]
> Agreed – fixing all flake8 violations would clearly be a waste of time.

> Redefinitions via list comprehensions is a real and recurring gotcha

Agreed that some flake8 warnings are more useful than others. Rather than have a PR for just one of the dangerous ones, perhaps it makes sense to have a single PR that adds all of the important checks?

The changes here seem fine, but I don't want to open the floodgates to "add flake8 waning ???? to Travis linting" PRs.

Marco is the test maintainer. Let's see what he thinks the correct approach is.

**Comment by practicalswift** [comment]
@jnewbery For the `flake8` violations we currently have in the code base I'd say that fixing and enabling Travis checking for `F841` (PR #12284: `local variable 'foo' is assigned to but never used`) and `F812` (this PR #12295: `list comprehension redefines 'foo' from line N `) would be enough.

If we want to go beyond that I'd suggest that we consider enabling the checks that we currently **do not** violate in our code base. Enabling them would guarantee that we will never introduce any new classes of violations in the future. The following `flake8` command does not emit any warnings on `master`:

```
$ git checkout master
$ git pull upstream master
$ flake8 --ignore=B,C,E,F,I,N,W --select=F402,F404,F406,F407,F601,F602,F621,F622,F631,\
       F701,F702,F703,F704,F705,F706,F707,F811,F822,F823,F831 .
$
```

These refer to:

```
F402 import module from line N shadowed by loop variable
F404 future import(s) name after other statements
F406 ‘from module import *’ only allowed at module level
F407 an undefined __future__ feature name was imported
F601 dictionary key name repeated with different values
F602 dictionary key variable name repeated with different values
F621 too many expressions in an assignment with star-unpacking
F622 two or more starred expressions in an assignment (a, *b, *c = d)
F631 assertion test is a tuple, which are always True
F701 a break statement outside of a while or for loop
F702 a continue statement outside of a while or for loop
F703 a continue statement in a finally block in a loop
F704 a yield or yield from statement outside of a function
F705 a return statement with arguments inside a generator
F706 a return statement outside of a function/method
F707 an except: block as not the last exception handler
F811 redefinition of unused name from line N
F822 undefined name name in __all__
F823 local variable name … referenced before assignment
F831 duplicate argument name in function definition
```

File: test/functional/rpc_fundrawtransaction.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def run_test(self):
         assert_raises_rpc_error(-1, "JSON value is not a string as expected", self.nodes[2].fundrawtransaction, rawtx, {'change_type': None})
         assert_raises_rpc_error(-5, "Unknown change type", self.nodes[2].fundrawtransaction, rawtx, {'change_type': ''})
         rawtx = self.nodes[2].fundrawtransaction(rawtx, {'change_type': 'bech32'})
-        tx  = self.nodes[2].decoderawtransaction(rawtx['hex'])
+        tx = self.nodes[2].decoderawtransaction(rawtx['hex'])
```
Comment (MarcoFalke) [inline test/functional/rpc_fundrawtransaction.py]: This should probably be renamed to dec_tx, like in all other sections.

**Review by maflcko — COMMENTED** [review]
Concept ACK on enabling the warning. However, I agree with @jnewbery that enabling and "fixing" each warning in a separate pull will generate too much noise.

**Comment by MarcoFalke** [comment]
>  Rather than have a PR for just one of the dangerous ones, perhaps it makes sense to have a single PR that adds all of the important checks?

Makes sense.

I will take a look at the list above at some point...

**Comment by practicalswift** [comment]
@MarcoFalke I've now renamed `tx` to `dec_tx` as requested. Would you mind re-reviewing? :-)

**Comment by MarcoFalke** [comment]
I read through the list of suggested lint warnings and I couldn't find an obvious wrong one. Also, since you claim that none of these are currently hit on master, I think it is fine to enable all of them in one go. Mind to repurpose this pull request for that?

**Comment by practicalswift** [comment]
@MarcoFalke Great idea! Now updated. Please review :-)

**Comment by MarcoFalke** [comment]
Cool. Thx.

utACK a9d0ebc26207b4771b7c240ca0c516debd330985

**Comment by MeshCollider** [comment]
Concept ACK / light review utACK, I think if we can add them without much churn then its sane, agree with above about fixing all 5000+ warnings though :)

**Comment by jnewbery** [comment]
utACK a9d0ebc26207b4771b7c240ca0c516debd330985

**Pull request merged**